### PR TITLE
Get *.js indentation from .editorconfig file

### PR DESCRIPTION
**Pull request by malandrew**
Fixes: https://github.com/uber/lint-trap/issues/39

Once this is accepted, I'll cut 0.5.0 (breaking api change since you need an .editorconfig file). Let me know if I should make editorconfig optional. My gut is no since .editorconfig does a lot to help multiple people working on the same project.

cc: @raynos

### CODE REVIEW

**Comment by Raynos** [comment]
@malandrew 

Semver is not optional. if you make a breaking change you got 1.0. 

In fact, the fact that you have any project that is <1.0 is silly, all modules should start at v1 or v2 out of the gate.

If you have some sort of emotional attachment to version "1.0" then forget that. Communicating with your users is always more important.

File: set-rules.js [hunk 1]
```diff
@@ -1,39 +1,12 @@
 'use strict';
-var detectIndent = require('detect-indent');
+require('array.prototype.find');
```
Comment (Raynos) [inline set-rules.js]: dont do that please. there is a nice `var find = require('find')` module.

File: set-rules.js [hunk 2]
```diff
@@ -46,30 +19,36 @@ function updateJSON(jsonPath, diffs) {
     jf.writeFileSync(jsonPath, content);
 }
 
-function setIndentRule(rootPath, firstFile, callback) {
-    if (firstFile === 'stdin') {
-        return process.nextTick(callback);
-    }
+function setIndentRule(rootPath, callback) {
+    findParentDir(rootPath, '.editorconfig', onEditorConfigDir);
```
Comment (Raynos) [inline set-rules.js]: please ship with a default .editorconfig if this doesnt exist.

**Comment by Raynos** [comment]
Please ship with a default editorconfig in the rc folder.

**Comment by malandrew** [comment]
Got it. 

Todo:
- Default .editorconfig or don't require it.
- Bump to version 1.0.0.

File: set-rules.js [hunk 3]
```diff
@@ -46,30 +19,36 @@ function updateJSON(jsonPath, diffs) {
...
     }
 
-    function findReferenceFileCallback(err, referenceFile) {
-        if (err) {
-            return callback(err);
+    function onEditorConfigParse(parseErr, parsed) {
+        if (parseErr) {
+            return callback(parseErr);
+        }
+        var jsRules = parsed.find(function getJsRuleSet(ruleset) {
```
Comment (lxe) [inline set-rules.js]: You can just do `[].filter()` or `[].some()` i think...

**Comment by malandrew** [comment]
@Raynos 

Let me know if this is okay now.

**Comment by malandrew** [comment]
Since I'm going to cut a 1.0.0 for this, I'm also resolving these issues:
- https://github.com/uber/lint-trap/issues/86
- https://github.com/uber/lint-trap/issues/82
- https://github.com/uber/lint-trap/issues/83

**Comment by malandrew** [comment]
Resolving https://github.com/uber/lint-trap/issues/76

**Comment by Raynos** [comment]
lgtm.

**Pull request merged**